File: python3_solution_set2/2605. Form Smallest Number From Two Digit Arrays.py
class Solution:
    def minNumber(self, nums1: List[int], nums2: List[int]) -> int:
        nums1.sort()
        nums2.sort()

        for x in nums1:
            if x in nums2:
                return x

        a = nums1[0]
        b = nums2[0]

        if a < b:
            return 10*a+b
        else:
            return 10*b + a

File: go_solution_set1/2605. Form Smallest Number From Two Digit Arrays.go
func minNumber(nums1 []int, nums2 []int) int {
    sort.Ints(nums1)
    sort.Ints(nums2)

    for _, val := range nums1{
        if has_element(nums2, val){
            return val
        }
    }

    if nums1[0] > nums2[0]{
        return nums2[0]*10 + nums1[0]
    } else {
        return nums2[0] + nums1[0]*10
    }
}

func has_element(arr []int, target int) bool{
    for _, x := range arr{
        if x == target{
            return true
        }
    }

    return false
}
